api4/cloud.go

```diff
@@ -195,7 +191,30 @@ func validateBusinessEmail(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// validate current userAdmin email
+	// if an email was sent as a body param, validate it and return wether is valid or not
+	bodyBytes, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		c.Err = model.NewAppError("Api4.requestCloudTrial", "api.cloud.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	var emailToValidate *model.ValidateBusinessEmailRequest
+	if err := json.Unmarshal(bodyBytes, &emailToValidate); err != nil {
+		c.Err = model.NewAppError("Api4.requestCloudTrial", "api.cloud.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if emailToValidate.Email != "" {
+		errValidatingEmail := c.App.Cloud().ValidateBusinessEmail(user.Id, emailToValidate.Email)
+		if errValidatingEmail != nil {
+			c.Err = model.NewAppError("Api4.valiateBusinessEmail", "api.cloud.request_error", nil, errValidatingEmail.Error(), http.StatusInternalServerError)
+			return
+		}
+		ReturnStatusOK(w)
+		return
+	}
+
```

Comment:
If the email is not valid, we still need to fall back on the workspace owner's email check below. If seems here it just returns an error without the check that was done previously

Also, I'm wondering if we should:

- Invert (or extract) the logic of first checking if the workspace owner has a valid business email. If they do, then we don't need to check anyone else and continue the trial request. If not, then we move to check the email of the current admin. 
- If we do extract the logic, we should keep this api call as just a dumb passthrough of validate any business email given 
- If we do not extract the logic from here, I propose we change the name of this api call to something that indicates we are doing more than just validating the given email - maybe `validateWorkspaceBusinessEmail` or `validateWorkspaceBusinessContact` 0/5

Reply:
the workflow is the following:
- The UI sends a first request to validate user admin email and customer creation used email: (no email is sent with the first api request), the user admin email is validated, in case of being an invalid email, the customer account used email is validated too. If both are invalid, the UI receives this response and it keeps that info in memory, so in case the user  wants to start the trial, the request email modal is prompted requesting for a valid business email, then the subsequent request to this api includes the entered email value, which will be validated too. Also, the changes included with this PR from the front-end add live validation of the business email providing inmediate feedback to the user.

Reply:
Do we have a ticket for this followup work? @pvev

Reply:
yes, this one https://mattermost.atlassian.net/browse/MM-45174